src/sinks/clickhouse/sink.rs

```diff
@@ -82,61 +77,11 @@ impl StreamSink<Event> for ClickhouseSink {
     }
 }
 
-struct ClickhouseRequestBuilder {
-    compression: Compression,
-    encoding: (Transformer, Encoder<Framer>),
-}
-
-impl RequestBuilder<(PartitionKey, Vec<Event>)> for ClickhouseRequestBuilder {
-    type Metadata = (PartitionKey, EventFinalizers);
-    type Events = Vec<Event>;
-    type Encoder = (Transformer, Encoder<Framer>);
-    type Payload = Bytes;
-    type Request = ClickhouseRequest;
-    type Error = std::io::Error;
-
-    fn compression(&self) -> Compression {
-        self.compression
-    }
-
-    fn encoder(&self) -> &Self::Encoder {
-        &self.encoding
-    }
-
-    fn split_input(
-        &self,
-        input: (PartitionKey, Vec<Event>),
-    ) -> (Self::Metadata, RequestMetadataBuilder, Self::Events) {
-        let (key, mut events) = input;
-
-        let finalizers = events.take_finalizers();
-        let builder = RequestMetadataBuilder::from_events(&events);
-        ((key, finalizers), builder, events)
-    }
-
-    fn build_request(
-        &self,
-        metadata: Self::Metadata,
-        request_metadata: RequestMetadata,
-        payload: EncodeResult<Self::Payload>,
-    ) -> Self::Request {
-        let (key, finalizers) = metadata;
-        ClickhouseRequest {
-            database: key.database,
-            table: key.table,
-            body: payload.into_payload(),
-            compression: self.compression,
-            finalizers,
-            metadata: request_metadata,
-        }
-    }
-}
-
 /// PartitionKey used to partition events by (database, table) pair.
 #[derive(Hash, Eq, PartialEq, Clone, Debug)]
-struct PartitionKey {
-    database: String,
-    table: String,
+pub struct PartitionKey {
+    pub(super) database: String,
+    pub(super) table: String,
```

Comment:
Could these be reversed? i.e. `pub(super)` on the struct itself and `pub` on the fields?